Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1186,9 +1214,67 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     m_backgroundWebView = nil;
 }
 
+WebExtensionContext::EventListenterTypeCountedSet WebExtensionContext::loadBackgroundPageListenersFromStorage()
+{
+    NSDictionary *extensionState = currentState();
+    NSCountedSet *savedListeners = [NSKeyedUnarchiver unarchivedObjectOfClasses:[NSSet setWithArray:@[ NSCountedSet.class, NSNumber.class ]] fromData:[extensionState objectForKey:backgroundPageListenersKey] error:nil];
+
+    EventListenterTypeCountedSet result;
+    for (NSNumber *entry in savedListeners)
+        result.add(static_cast<WebExtensionEventListenerType>(entry.unsignedIntValue), [savedListeners countForObject:entry]);
+
+    return result;
+}
+
+NSString *WebExtensionContext::lastSeenUniqueIdentifier()
+{
+    return [m_state objectForKey:lastSeenUniqueIdentifierKey];
+}
+
+void WebExtensionContext::saveBackgroundPageListenersToStorage()
+{
+    if (backgroundContentURL().isEmpty() || isPersistent())
+        return;
+
+    NSMutableDictionary *extensionState = [currentState() mutableCopy];
+    if (!extensionState.count)
+        return;
+
+    NSCountedSet *listeners = [NSCountedSet set];
+    for (auto& entry : m_backgroundPageListeners)
+        [listeners addObject:@(static_cast<uint8_t>(entry.key))];
+
+    NSData *newBackgroundPageListenersAsData = [NSKeyedArchiver archivedDataWithRootObject:listeners requiringSecureCoding:YES error:nil];
+    NSData *savedBackgroundPageListenersAsData = [extensionState objectForKey:backgroundPageListenersKey];
+    extensionState[backgroundPageListenersKey] = newBackgroundPageListenersAsData;
```

Comment:
Use `m_state.get()[backgroundPageListenersKey]`

Reply:
I decided to do `[m_state objectForKey:backgroundPageListenersKey];`. Is there a preference of doing one over the other?